src/NJsonSchema.SourceGenerators.CSharp/NJsonSchema.SourceGenerators.CSharp.csproj

```diff
@@ -0,0 +1,74 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```

Comment:
My main concern is whether we should create a separate project for the source generator or add them to the existing `NJsonSchema.CodeGeneration.CSharp`. Below are my thoughts.
## Add Source Generator to new `NJsonSchema.SourceGenerators.CSharp`
### Pros:

- Clients will install that package only if they know they need to use a source generator.
- Easier to maintain as source generator projects require additional configuration (`IsRoslynComponent`, exclude NU5128 warning, etc.)
### Cons:
- The names of the NuGet packages could be confusing. `NJsonSchema.CodeGeneration.CSharp` and `NJsonSchema.SourceGenerators.CSharp` may sound a bit synonimical.
## Add Source Generator to existing `NJsonSchema.CodeGeneration.CSharp`
### Pros:
- We don't introduce a new NuGet package.
- By installing `NJsonSchema.CodeGeneration.CSharp` client can use either classes from the library or/and Source Generator.
### Cons: 
- It'll automatically install the analyzer when the client does a package upgrade. (However, the analyzer will not do much without additional configuration. Not sure if this is an issue).
- The project might be harder to maintain as it will be a mix of regular code and code that will go to the source generator. (Frankly speaking, I didn't try that approach at all, but in theory it should work).